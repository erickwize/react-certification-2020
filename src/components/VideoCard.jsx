import React from 'react';
import styled from 'styled-components';

const VideoCard = ({ data }) => {
  const formattedDate = new Date(data.snippet.publishedAt);
  const ye = new Intl.DateTimeFormat('en', { year: 'numeric' }).format(formattedDate);
  const mo = new Intl.DateTimeFormat('en', { month: 'short' }).format(formattedDate);
  const da = new Intl.DateTimeFormat('en', { day: '2-digit' }).format(formattedDate);

  return (
    <Card data-testid={data.etag}>
      <header>
        <img src={data.snippet.thumbnails.medium.url} alt={data.snippet.title} />
        <div>{data.snippet.title}</div>
      </header>
      {data.snippet.description !== '' ? (
        <div>{data.snippet.description}</div>
      ) : (
        <div>{`${data.snippet.channelTitle} - ${da} ${mo}, ${ye}`}</div>
      )}
    </Card>
  );
};

export default VideoCard;

const Card = styled.div`
  overflow: hidden;
  border-radius: 5px;
  box-shadow: 0px 0px 15px -5px rgba(0, 0, 0, 0.75);

  header {
    position: relative;
    width: 100%;
    max-height: 200px;
    overflow: hidden;
    cursor: pointer;
    &:hover {
      img {
        width: 175%;
        object-position: -7.5em -2em;
      }
    }

    img {
      width: 150%;
      transition: 0.2s;
      object-fit: cover;
      object-position: -5em -1em;
    }

    div {
      position: absolute;
      bottom: 0;
      padding: 3%;
      background-image: linear-gradient(0deg, #000 0%, transparent 100%);
      width: 100%;
      color: white;
      text-shadow: #000 0px 0px 5px;
    }
  }

  & > div {
    padding: 3%;
    font-size: 0.7em;
  }
`;
