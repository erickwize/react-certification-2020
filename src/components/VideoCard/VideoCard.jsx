import React from 'react';
import { Link } from 'react-router-dom';
import VideoBox from './VideoCard.styled';

function VideoCard({ data, videoList }) {
  const { title, description } = data?.snippet;
  const source = data?.snippet.thumbnails.default.url;
  const videoId = data?.id?.videoId;
  return (
    <VideoBox>
      <Link
        to={{
          pathname: `/video/${videoId}`,
          data: { data, videoList },
        }}
      >
        <img src={source} alt={title} />
        <h3>{title}</h3>
        <p data-testid="paragraph">{description}</p>
      </Link>
    </VideoBox>
  );
}

export default VideoCard;
