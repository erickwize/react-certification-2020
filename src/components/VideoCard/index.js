import React from 'react'
import styled from 'styled-components'

const VideoCardStyled = styled.div`
	width: 18%;
	margin: 20px 10px;
	border-radius: 5px;
	overflow: hidden;
	@media only screen and (max-width: 1400px) {
		width: 23%;
	}
	@media only screen and (max-width: 1100px) {
		width: 30%;
	}
	@media only screen and (max-width: 768px) {
		width: 45%;
	}
	@media only screen and (max-width: 475px) {
		width: 100%;
	}
`
const VideoThumbnail = styled.img`
	width: 100%;
	height: 120px;
	object-fit: cover;
	display: block;
`
const VideoInfo = styled.div``

const VideoTitle = styled.span`
	font-size: 14px;
	font-weight: 800;
	color: #444;
	padding: 5px;
	display: block;
`
const VideoDescription = styled.p`
	color: #888;
	font-size: 12px;
	font-style: italic;
	margin: 0;
	padding: 0 5px;
`

export default function VideoCard({ item }) {
	return (
		<VideoCardStyled>
			<VideoThumbnail src={item.snippet.thumbnails.default.url} alt="thumbnail" />
			<VideoInfo>
				<VideoTitle>{item.snippet.title}</VideoTitle>
				<VideoDescription>{item.snippet.description}</VideoDescription>
			</VideoInfo>
		</VideoCardStyled>
	)
}
