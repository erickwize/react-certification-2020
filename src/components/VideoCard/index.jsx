import React from 'react';

const VideoCard = (props) => (
  <div id="{props.key}" className="max-w-sm rounded overflow-hidden shadow-lg">
    <img src={props.urlImage} alt="" className="w-full" />
    <div className="px-2 py-4">
      <div className="font-bold text-black text-left font-sans mb-2">{props.title}</div>
      <div className="text-sm text-left font-sans">{props.description}</div>
      <div className="flex justify-between bg-gray-400">
        <div className="flex-shrink md:flex-shrink-0 text-xs self-start p-1">
          {props.publishedDate}
        </div>
        <div className="md:flex-shrink-0 bg-red-500 font-serif text-xs italic text-white p-1 self-end">
          {props.channel}
        </div>
      </div>
    </div>
  </div>
);

export default VideoCard;
