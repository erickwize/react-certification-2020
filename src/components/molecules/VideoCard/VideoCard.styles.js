import styled from 'styled-components';

export const CardContainer = styled.div`
  display: block;
  flex-direction: column;
  height: fit-content;
  width: 320px;
  height: 320px;
  background-color: rgba(185, 134, 217, 0.2);
  border-radius: 6px;
  margin-bottom: 30px;
  overflow: hidden;
  box-shadow: rgba(50, 44, 78, 0.15) 0px 5px 15px 0px;
  &:hover {
    position: 'absolute';
    opacity: 0.6;
    box-shadow: rgba(254, 223, 3, 0.12) 0px 2px 4px 0px,
      rgba(254, 223, 3) 0px 2px 16px 0px;
  }
  @media screen and (max-width: 812px) {
    width: 95%;
    max-width: 320px;
    height: 320px;
  }
  cursor: not-allowed;
`;

// box-shadow: rgba(149, 157, 165, 0.2) 0px 8px 24px;

export const TextContainer = styled.div`
  margin: 0.5rem;
`;
