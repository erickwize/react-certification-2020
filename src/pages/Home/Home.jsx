import React, { useRef } from 'react';

import './Home.css';
import VideoList from '../../components/VideoList/VideoList';

function Home() {
  const sectionRef = useRef(null);

  return (
    <div ref={sectionRef}>
      <h2 className="title">Welcome to the Challenge!</h2>
      <VideoList />
    </div>
  );
}

export default Home;
