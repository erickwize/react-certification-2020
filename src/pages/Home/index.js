import React from 'react'
import styled from 'styled-components'
import VideosList from '../../components/VideosList'

const HomeStyled = styled.div`
	padding: 20px;
`
const Title = styled.h1`
	text-align: center;
`

const Home = () => {
	return (
		<HomeStyled>
			<Title>Results</Title>
			<VideosList />
		</HomeStyled>
	)
}

export default Home
