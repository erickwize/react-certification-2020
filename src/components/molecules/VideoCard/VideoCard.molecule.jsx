import React from 'react';
import { CardContainer, TextContainer } from './VideoCard.styles';
import { Thumbnail, VideoInformation } from '../../atoms';

function VideoCard({ videos }) {
  return (
    <CardContainer>
      <Thumbnail bg={<img src={videos.snippet.thumbnails.medium.url} />} />
      <TextContainer>
        <VideoInformation
          title={videos.snippet.title}
          description={videos.snippet.description}
        />
      </TextContainer>
    </CardContainer>
  );
}

export default VideoCard;
