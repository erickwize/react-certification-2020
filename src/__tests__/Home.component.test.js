import React from 'react';
import renderer from 'react-test-renderer';
import { mount, render, configure } from 'enzyme';
import Home from '../pages/Home';
import VideoCard from '../components/VideoCard';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

it('renders correctly', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('check title by default', () => {
  const title = mount(<Home />);
  expect(title.find('h2').text()).toEqual('Mini Challenge 1');
});

it('renders VideoCard component', () => {
  const wrapper = mount(<Home />);
  expect(wrapper.containsMatchingElement(<VideoCard />)).toEqual(true);
});
