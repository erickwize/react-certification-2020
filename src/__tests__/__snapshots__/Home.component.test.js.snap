// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="sc-jSgupP bGUhEE"
>
  <h2
    className="sc-gKsewC hHAcPG"
  >
    Mini Challenge 1
  </h2>
  <div
    className="sc-eCssSg hUZFej"
  >
    <div
      className="sc-bdfBwQ jqRnJH"
    >
      <div
        className="sc-gsTCUz gAjrqo"
      />
      <h3
        className="sc-dlfnbm udFbv"
      >
        Wizeline
      </h3>
      <p
        className="sc-hKgILt gbWVGw"
      >
        Wizeline transforms how teams build technology. Its customers accelerate the delivery of innovative products with proven solutions, which combine Wizeline's ...
      </p>
    </div>
    <div
      className="sc-bdfBwQ jqRnJH"
    >
      <div
        className="sc-gsTCUz dXfHfQ"
        height={180}
        width={320}
      />
      <h3
        className="sc-dlfnbm udFbv"
      >
        Video Tour | Welcome to Wizeline Guadalajara
      </h3>
      <p
        className="sc-hKgILt gbWVGw"
      >
        Follow Hector Padilla, Wizeline Director of Engineering, for a lively tour of our office. In 2018, Wizeline opened its stunning new office in Guadalajara, Jalisco, ...
      </p>
    </div>
    <div
      className="sc-bdfBwQ jqRnJH"
    >
      <div
        className="sc-gsTCUz cohgRu"
        height={180}
        width={320}
      />
      <h3
        className="sc-dlfnbm udFbv"
      >
        Wizeline Guadalajara | Bringing Silicon Valley to Mexico
      </h3>
      <p
        className="sc-hKgILt gbWVGw"
      >
        Wizeline continues to offer a Silicon Valley culture in burgeoning innovation hubs like Mexico and Vietnam. In 2018, our Guadalajara team moved into a ...
      </p>
    </div>
    <div
      className="sc-bdfBwQ jqRnJH"
    >
      <div
        className="sc-gsTCUz fQDrxW"
        height={180}
        width={320}
      />
      <h3
        className="sc-dlfnbm udFbv"
      >
        Wizeline hace sentir a empleados como en casa
      </h3>
      <p
        className="sc-hKgILt gbWVGw"
      >
        En el 2014, Bismarck fundó Wizeline, compañía tecnológica que trabaja con los corporativos ofreciendo una plataforma para que desarrollen software de forma ...
      </p>
    </div>
    <div
      className="sc-bdfBwQ jqRnJH"
    >
      <div
        className="sc-gsTCUz eSFeCr"
        height={180}
        width={320}
      />
      <h3
        className="sc-dlfnbm udFbv"
      >
        We Are Wizeline
      </h3>
      <p
        className="sc-hKgILt gbWVGw"
      >
        Engineering a better tomorrow. Wizeline is a global software development company that helps its clients solve their biggest challenges with design and ...
      </p>
    </div>
    <div
      className="sc-bdfBwQ jqRnJH"
    >
      <div
        className="sc-gsTCUz yHhBV"
        height={180}
        width={320}
      />
      <h3
        className="sc-dlfnbm udFbv"
      >
        Wizeline Thrives in Mexico City
      </h3>
      <p
        className="sc-hKgILt gbWVGw"
      >
        Our vibrant Mexico City office is home to agile software engineers, talented UX designers, and brilliant data scientists. Learn about the rich history of Mexico City.
      </p>
    </div>
    <div
      className="sc-bdfBwQ jqRnJH"
    >
      <div
        className="sc-gsTCUz bFpCoH"
        height={180}
        width={320}
      />
      <h3
        className="sc-dlfnbm udFbv"
      >
        Wizeline
      </h3>
      <p
        className="sc-hKgILt gbWVGw"
      >
        El plan de Wizeline, una empresa de inteligencia artificial, para ayudar a crecer la comunidad de ciencia de datos en CDMX y todo el país, a través de cursos ...
      </p>
    </div>
    <div
      className="sc-bdfBwQ jqRnJH"
    >
      <div
        className="sc-gsTCUz jPQXEE"
        height={180}
        width={320}
      />
      <h3
        className="sc-dlfnbm udFbv"
      >
        A Day in the Life of an Engineering Manager at Wizeline
      </h3>
      <p
        className="sc-hKgILt gbWVGw"
      >
        Fernando Espinoza shares his experience working as an engineering manager at Wizeline and mentoring other engineers. Learn about Fernando's passions ...
      </p>
    </div>
    <div
      className="sc-bdfBwQ jqRnJH"
    >
      <div
        className="sc-gsTCUz fiNMqz"
        height={180}
        width={320}
      />
      <h3
        className="sc-dlfnbm udFbv"
      >
        Wizeline Technical Writing Academy | Featuring Eduardo Ocejo
      </h3>
      <p
        className="sc-hKgILt gbWVGw"
      >
        
      </p>
    </div>
    <div
      className="sc-bdfBwQ jqRnJH"
    >
      <div
        className="sc-gsTCUz cjaBDm"
        height={180}
        width={320}
      />
      <h3
        className="sc-dlfnbm udFbv"
      >
        Silicon Valley en México
      </h3>
      <p
        className="sc-hKgILt gbWVGw"
      >
        Empresas de Silicon Valley buscan establecerse en México por el gran talento que hay en nuestro país. Es una investigación de Roberto Domínguez.
      </p>
    </div>
    <div
      className="sc-bdfBwQ jqRnJH"
    >
      <div
        className="sc-gsTCUz fdzVwN"
        height={180}
        width={320}
      />
      <h3
        className="sc-dlfnbm udFbv"
      >
        Why Wizeline? featuring Juan Pablo Villa in Mexico City
      </h3>
      <p
        className="sc-hKgILt gbWVGw"
      >
        Juan Pablo, known as Gianpa at Wizeline, is a software engineer in our Mexico City office. Gianpa focuses on Android apps, is an integral part of our culture, ...
      </p>
    </div>
    <div
      className="sc-bdfBwQ jqRnJH"
    >
      <div
        className="sc-gsTCUz CGBIC"
        height={180}
        width={320}
      />
      <h3
        className="sc-dlfnbm udFbv"
      >
        Wizeline's 2020 Year in Review
      </h3>
      <p
        className="sc-hKgILt gbWVGw"
      >
        There's no doubt that 2020 has been an unprecedented year. However, amidst all the chaos, we achieved remarkable growth in various areas of our business.
      </p>
    </div>
    <div
      className="sc-bdfBwQ jqRnJH"
    >
      <div
        className="sc-gsTCUz jViRqD"
        height={180}
        width={320}
      />
      <h3
        className="sc-dlfnbm udFbv"
      >
        Why Wizeline? featuring Hugo Lopez in Mexico City
      </h3>
      <p
        className="sc-hKgILt gbWVGw"
      >
        
      </p>
    </div>
    <div
      className="sc-bdfBwQ jqRnJH"
    >
      <div
        className="sc-gsTCUz eauVPe"
        height={180}
        width={320}
      />
      <h3
        className="sc-dlfnbm udFbv"
      >
        Why Wizeline? (We're Hiring in Mexico!)
      </h3>
      <p
        className="sc-hKgILt gbWVGw"
      >
        A quick look at why people join Wizeline, what motivates us as a team and what it's like to work in our Guadalajara office. Learn more and apply here: ...
      </p>
    </div>
    <div
      className="sc-bdfBwQ jqRnJH"
    >
      <div
        className="sc-gsTCUz bfVCKP"
        height={180}
        width={320}
      />
      <h3
        className="sc-dlfnbm udFbv"
      >
        Why Wizeline? featuring Oswaldo Herrera in Mexico City
      </h3>
      <p
        className="sc-hKgILt gbWVGw"
      >
        Oswaldo is a software engineering in Wizeline's Mexico City office. He joined Wizeline because of the camaraderie and deep sense of commitment of our teams.
      </p>
    </div>
    <div
      className="sc-bdfBwQ jqRnJH"
    >
      <div
        className="sc-gsTCUz eBbhGn"
        height={180}
        width={320}
      />
      <h3
        className="sc-dlfnbm udFbv"
      >
        Caso de Éxito AWS: Wizeline [Spanish]
      </h3>
      <p
        className="sc-hKgILt gbWVGw"
      >
        Central de socios de APN - https://amzn.to/2S7tIXM Fundada en 2014, Wizeline es una compañía joven e innovadora que nació en la nube para ofrecer soporte ...
      </p>
    </div>
    <div
      className="sc-bdfBwQ jqRnJH"
    >
      <div
        className="sc-gsTCUz ekzobu"
        height={180}
        width={320}
      />
      <h3
        className="sc-dlfnbm udFbv"
      >
        Wizeline's 2019 Year in Review
      </h3>
      <p
        className="sc-hKgILt gbWVGw"
      >
        
      </p>
    </div>
    <div
      className="sc-bdfBwQ jqRnJH"
    >
      <div
        className="sc-gsTCUz lcGotP"
        height={180}
        width={320}
      />
      <h3
        className="sc-dlfnbm udFbv"
      >
        [1 of 2] Wizeline CEO shares career lessons from Google
      </h3>
      <p
        className="sc-hKgILt gbWVGw"
      >
        Founder & CEO Bismarck Lepe on growth opportunities at Wizeline and his career-path experience as an early Google employee. Join our team!
      </p>
    </div>
    <div
      className="sc-bdfBwQ jqRnJH"
    >
      <div
        className="sc-gsTCUz MoMOr"
        height={180}
        width={320}
      />
      <h3
        className="sc-dlfnbm udFbv"
      >
        Wizeline Data Engineering featuring Tania Reyes
      </h3>
      <p
        className="sc-hKgILt gbWVGw"
      >
        Tania discovered her interest in Big Data while working at Wizeline and took Wizeline Academy courses to skill up and join the data team. Now, she works on ...
      </p>
    </div>
    <div
      className="sc-bdfBwQ jqRnJH"
    >
      <div
        className="sc-gsTCUz IhPEh"
        height={180}
        width={320}
      />
      <h3
        className="sc-dlfnbm udFbv"
      >
        Wizeline Querétaro | Mexico's New Knowledge Economy (We're hiring!)
      </h3>
      <p
        className="sc-hKgILt gbWVGw"
      >
        A small but mighty (and growing) team, the Queretaro crew has taken ownership of growing the office and brand, speaking at university events, hosting tech ...
      </p>
    </div>
    <div
      className="sc-bdfBwQ jqRnJH"
    >
      <div
        className="sc-gsTCUz dCtBuF"
        height={180}
        width={320}
      />
      <h3
        className="sc-dlfnbm udFbv"
      >
        Miriam Godinez | Women in Leadership at Wizeline
      </h3>
      <p
        className="sc-hKgILt gbWVGw"
      >
        Science and technology always caught Miriam's attention. One of her ultimate goals as a Senior Engineer Manager and Lead from the Mobile Team at Wizeline ...
      </p>
    </div>
    <div
      className="sc-bdfBwQ jqRnJH"
    >
      <div
        className="sc-gsTCUz esfOwz"
        height={180}
        width={320}
      />
      <h3
        className="sc-dlfnbm udFbv"
      >
        Welcome Back to Wizeline Vietnam | Extended Version
      </h3>
      <p
        className="sc-hKgILt gbWVGw"
      >
        Thanks to swift government action, the COVID-19 situation in Vietnam has reached a point where businesses are able to return to work and reopen offices.
      </p>
    </div>
    <div
      className="sc-bdfBwQ jqRnJH"
    >
      <div
        className="sc-gsTCUz jCNWOr"
        height={180}
        width={320}
      />
      <h3
        className="sc-dlfnbm udFbv"
      >
        How does Wizeline work?
      </h3>
      <p
        className="sc-hKgILt gbWVGw"
      >
        Wizeline builds teams with a mix of technical and non-technical talent to deliver better products, faster. Learn more about our consulting services: ...
      </p>
    </div>
    <div
      className="sc-bdfBwQ jqRnJH"
    >
      <div
        className="sc-gsTCUz eAxWd"
        height={180}
        width={320}
      />
      <h3
        className="sc-dlfnbm udFbv"
      >
        12 Wishes from Wizeline | Happy Holidays 2019
      </h3>
      <p
        className="sc-hKgILt gbWVGw"
      >
        
      </p>
    </div>
    <div
      className="sc-bdfBwQ jqRnJH"
    >
      <div
        className="sc-gsTCUz iZyGqb"
        height={180}
        width={320}
      />
      <h3
        className="sc-dlfnbm udFbv"
      >
        Why Wizeline?
      </h3>
      <p
        className="sc-hKgILt gbWVGw"
      >
        Hear from our employees directly about what excites them about their roles here at Wizeline. Wizeline wants to hire the best and the brightest to accelerate their ...
      </p>
    </div>
  </div>
</section>
`;
